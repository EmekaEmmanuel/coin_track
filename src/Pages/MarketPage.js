import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchMarkets } from '../redux/Market/marketSlice';
import Market from '../Components/Market';
import Nav from '../Components/Nav';
import Footer from '../Components/Footer';

function MarketPage() {
  const myMarkets = useSelector((state) => state.market.markets);
  const dispatch = useDispatch();
  const { id } = useParams();

  useEffect(() => {
    let active = true;
    (async () => {
      if (active) {
        dispatch(fetchMarkets(id));
      }
    })();
    return () => {
      active = false;
    };
  }, [dispatch, id]);

  return (
    <div className="app_container">
      <Nav />
      <Market myMarkets={myMarkets} />
      <Footer />
    </div>
  );
}

export default MarketPage;
