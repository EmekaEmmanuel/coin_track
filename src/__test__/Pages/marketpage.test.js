import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import MarketPage from '../../Pages/MarketPage';

it('Market page renders correctly', () => {
  const myMarkets = store.getState().market.markets;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <MarketPage myMarkets={myMarkets} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
