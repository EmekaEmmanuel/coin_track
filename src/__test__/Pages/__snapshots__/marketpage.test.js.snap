// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Market page renders correctly 1`] = `
<div
  className="app_container"
>
  <div
    className="app_container"
  >
    <header
      className="home_header_box header_line"
    >
      <div
        className="home_header "
      >
        <div
          className="logo_box"
        >
          <figure>
            <img
              alt=""
              className="logo_img"
              src="crypto_image.png"
            />
          </figure>
          <h2
            className="nav_font1"
          >
            monie track
          </h2>
        </div>
        <nav
          className="nav_tag"
        >
          <li
            className="active nav_item"
          >
            <a
              activeClassName="active"
              aria-current="page"
              className="nav_font2 td_none active"
              href="/"
              onClick={[Function]}
            >
              Track 
            </a>
          </li>
          <li
            className="active nav_item"
          >
            <a
              activeClassName="active"
              aria-current="page"
              className="nav_font2 td_none active"
              href="/"
              onClick={[Function]}
            >
              Happenings 
            </a>
          </li>
          <li
            className="active nav_item"
          >
            <a
              activeClassName="active"
              className="nav_font2 td_none"
              href="/ChartPage"
              onClick={[Function]}
            >
              Chart 
            </a>
          </li>
        </nav>
        <div
          className="notify_box"
        >
          <button
            type="button"
          >
             
            <img
              alt=""
              src="closepopup.png"
            />
          </button>
          <button
            type="button"
          >
             
            <img
              alt=""
              src="closepopup.png"
            />
          </button>
        </div>
        <div
          className="hamburger_container"
        >
          <figure>
            <img
              alt="Hamburger_menu"
              src="hamburger.png"
            />
          </figure>
        </div>
      </div>
    </header>
  </div>
  <div
    className="app_container"
  >
    <main
      className="market_main_tag"
    >
      <section
        className="market_section1"
      >
        <p
          className="market_font1"
        >
          BE THE FIRST TO KNOW ABOUT MARKET CHANGES
        </p>
        <h1
          className="market_font2"
        >
          Get an Overview on multi market exchanges
        </h1>
        <p
          className="market_font3"
        >
          Market multiplicity enables users to manage and trade their assets with greater transparency in a permissionless environment with a simplified and straightforward interface.
        </p>
      </section>
      <section
        className="market_section2"
      />
    </main>
  </div>
  <div
    className="app_container"
  >
    <footer
      className="home_footer"
    >
      <article
        className="home_footer_article"
      >
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Company
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Trade
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Help
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Products
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
      </article>
      <div
        className="footer_copyright"
      >
        Â© 2022 monie track. All rights reserved.
      </div>
    </footer>
  </div>
</div>
`;
